packages/jest-expect/src/index.ts

```diff
@@ -6,27 +6,30 @@
  */
 
 import type {Config} from '@jest/types';
-import {Expect, expect} from 'expect';
+import {expect} from 'expect';
 import {
   addSerializer,
   toMatchInlineSnapshot,
   toMatchSnapshot,
   toThrowErrorMatchingInlineSnapshot,
   toThrowErrorMatchingSnapshot,
 } from 'jest-snapshot';
+import type {JestExpect} from './types';
 
-export default function jestExpect(
-  config: Pick<Config.GlobalConfig, 'expand'>,
-): Expect {
-  expect.setState({expand: config.expand});
-  expect.extend({
+export type {JestExpect} from './types';
+
+export function createJestExpect({expand}: Config.GlobalConfig): JestExpect {
+  const jestExpect = expect as JestExpect;
+
+  jestExpect.setState({expand});
+  jestExpect.extend({
```

Comment:
Here `expect.extend()` does include all snapshot matchers as asymmetric matchers as well. Nor documentation, neither types do reflect that. Hm.. Fine for now, but seems like `expect.extend()` must be improved. I was thinking about it, but let’s leave it for another PR.

Reply:
yeah, they cannot be used as asymmetric matchers, so we should figure something out before shipping v28 stable. Perfectly fine to punt for now, of course 🙂

Reply:
My idea is to introduce second argument with `enum` flags in a pattern [used by TypeScript](https://github.com/microsoft/TypeScript/blob/b9a06e515ff4ffaaf5eba87fa49f169fb532c7b5/src/compiler/types.ts#L800-L809). It should default to "add matcher everywhere" (current behaviour), but also allow, for example, to add some matcher function only as inverted asymmetric matcher. Having only one `flags` argument might make it possible to use function overloads which would infer types of matcher function and would add them to correct parts of `Expect` type. Uff.. Big plans. All might fail, but I will try first ;D

Reply:
exciting! 😀